backend/z_index.js

```diff
@@ -0,0 +1,153 @@
+const cors = require("cors");
+const app = express();
+const bodyParser = require("body-parser");
+const { request } = require('http');
+//middleware
+app.use(cors({
+  origin: '*'
+}));
+app.use(express.json());
+//app.use(bodyParser.urlencoded({ extended: false }))
+//app.use(bodyParser.json());
+//app.use(express.static('public'))
+// app.use((req, res, next) => {
+//   res.header('Access-Control-Allow-Origin', '*');
+//   res.header('Access-Control-Allow-Headers', "Origin, X-Requested-With, Content-Type, Accept");
+//   next();
+// });
+var code = "";
+var duration = null;
+//endpoints 
+app.get('/members', async (req, res) => {
+  const members = await getMembers()
+  res.json(members)
+})
+app.get('/forms', async (req, res) => {
+  const forms = await getForms()
+  res.json(forms)
+})
+app.get('/absences', async (req, res) => {
+  const absences = await getAbsenceForms();
+  res.json(absences);
+})
+app.post('/meeting', (req, res) => {
+  if (req.body == undefined) {
+    return res.json({
+      msg: "Error: parameter not defined",
+      data: {}
+    });
+  }
+  if (code != "") {
+    return res.json({
+      msg: "Meeting currently active. Cancel other meeting before starting new one",
+      data: { code }
+    })
+  }
+  const body = req.body
+  console.log(body);
+  code = body.code;
+  duration = body.duration;
+  console.log(code);
+  console.log(duration);
+  return res.json({
+    msg: "Success",
+    data: {
+      code,
+      duration
+    }
+  });
+});
+app.listen(PORT, console.log(`Server started on port ${PORT}`))
+
+//variables that will be sent in from frontend (hardcoded for now)
+var meetingType = "Engineering";
+
+const updateMembersData = async () => {
+  const { Client } = require('@notionhq/client');
+  const notion = new Client({ auth: process.env.NOTION_KEY });
+
+  //TODO: MAKE MORE EFFICIENT- should only do this computation 
+  //once and figure out how to store it
+  const absences = await getAbsenceForms();
+  const present = await getForms();
+
+  constAbsencesSet = new Set();
+  for (var i = 0; i < absences.length; i++) {
+    var obj = absences[i];
+    if (obj != undefined) {
+      constAbsencesSet.add(obj.name);
+    }
+  }
+  //console.log(constAbsencesSet);
+  constPresentSet = new Set();
+  for (var i = 0; i < present.length; i++) {
+    var obj = present[i];
+    if (obj != undefined) {
+      constPresentSet.add(obj.name);
+    }
+  }
+  //console.log(constPresentSet);
+
+  const members = await getMembers();
+  for (var i = 0; i < members.length; i++) {
+    var obj = members[i];
+    //console.log(obj);
+    if (obj != undefined) {
+      //ensures only general and team-specific meetings counted for attendance
+      if (meetingType === "General" || obj.team === meetingType) {
+        if (constPresentSet.has(obj.name)) {
+          //console.log("present " + obj.name);
+          //function that updates members data base with 1 more meeting attended
+          (async () => {
```

Comment:
Wait isn't the entire code wrapped with async already? Why are we calling async again?

Reply:
@zshipchandler any idea on this?

Reply:
the repeated async call was unnecessary- I fixed it in my new pull request